synthesizer/src/vm/helpers/cost.rs

```diff
@@ -43,12 +45,17 @@ pub fn deployment_cost<N: Network>(deployment: &Deployment<N>) -> Result<(u64, (
         .ok_or(anyhow!("The namespace cost computation overflowed for a deployment"))?
         .saturating_mul(1_000_000); // 1 microcredit = 1e-6 credits.
 
+    // Compute the synthesis cost in credits
+    let synthesis_cost = num_constraints * N::SYNTH_FEE_MULTIPLIER;
+
     // Compute the total cost in microcredits.
     let total_cost = storage_cost
         .checked_add(namespace_cost)
+        .ok_or(anyhow!("The total cost computation overflowed for a deployment"))?
+        .checked_add(synthesis_cost)
```

Comment:
`Option::and_then` could be used here to avoid duplicating the error message

Reply:
457334845